centreon-open-tickets/src/CentreonOpenTickets/Providers/Application/UseCase/FindProviders.php

```diff
@@ -0,0 +1,110 @@
+use CentreonOpenTickets\Providers\Application\Exception\ProviderException;
+use CentreonOpenTickets\Providers\Application\Repository\ReadProviderRepositoryInterface;
+use CentreonOpenTickets\Providers\Domain\Model\Provider;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+
+final class FindProviders
+{
+    use LoggerTrait;
+    public const ROLE_CONFIGURATION_OPEN_TICKETS = 'ROLE_CONFIGURATION_NOTIFICATIONS_RULES_RW';
+
+    /**
+     * @param ContactInterface $contact
+     * @param RequestParametersInterface $requestParameters
+     * @param ReadProviderRepositoryInterface $repository
+     */
+    public function __construct(
+        private ContactInterface $contact,
+        private RequestParametersInterface $requestParameters,
+        private ReadProviderRepositoryInterface $repository
+    ) {
+
+    }
+
+    /**
+     * @param FindProvidersPresenterInterface $presenter
+     */
+    public function __invoke(FindProvidersPresenterInterface $presenter): void
+    {
+        if (! $this->isAuthorized()) {
+            $this->error(
+                "User doesn't have sufficient rights to get ticket providers information",
+                [
+                    'user_id' => $this->contact->getId(),
+                ]
+            );
+            $presenter->presentResponse(
+                new ForbiddenResponse(ProviderException::listingNotAllowed()->getMessage())
+            );
+
+            return;
+        }
+
+        try {
+            $providers = $this->repository->findByRequestParameters($this->requestParameters);
+            $presenter->presentResponse($this->createResponse($providers));
+        } catch (RequestParametersTranslatorException $ex) {
+            $presenter->presentResponse(new ErrorResponse($ex->getMessage()));
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (\Throwable $exception) {
+            $presenter->presentResponse(new ErrorResponse(ProviderException::errorWhileListingProviders()));
+            $this->error($exception->getMessage(), ['trace' => $exception->getTraceAsString()]);
+        }
+    }
+
+    /**
+     * @param Provider[] $providers
+     *
+     * @return FindProvidersResponse
+     */
+    private function createResponse(array $providers): FindProvidersResponse
+    {
+        $response = new FindProvidersResponse();
+        $response->providers = array_map(
+            static fn (Provider $provider): ProviderDto => new ProviderDto(
+                id: $provider->getId(),
+                name: $provider->getName(),
+                type: $provider->getType(),
+                isActivated: $provider->isActivated()
+            ),
+            $providers
+        );
+
+        return $response;
+    }
+
+    private function isAuthorized(): bool
+    {
+        return $this->contact->isAdmin() || $this->contact->hasTopologyRole(self::ROLE_CONFIGURATION_OPEN_TICKETS);
```

Comment:
An admin has all the roles
```suggestion
        return $this->contact->hasTopologyRole(self::ROLE_CONFIGURATION_OPEN_TICKETS);
```